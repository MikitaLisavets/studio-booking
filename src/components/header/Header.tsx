import styles from './Header.module.scss';
import React, { MouseEvent } from 'react';
import useShallowEqualSelector from '../../hooks/useShallowEqualSelector';
import Display from '../../utils/Display';
import L from '../../utils/locale';
import { Link } from 'react-router-dom';
import { getUser } from '../../selectors/selectors';
import { SIGN_UP_ROUTE, LOGIN_ROUTE, LOGOUT_ROUTE } from '../../constants/navigation';
import { useDispatch } from 'react-redux';
import { logout } from '../../actions/Actions';


export default function Header(): JSX.Element {
  const dispatch = useDispatch();
  const user = useShallowEqualSelector(getUser);

  function handleLogoutClick(event: MouseEvent): void {
    event.preventDefault();
    dispatch(logout());
  }

  return (
    <div className={styles.header}>
      <button className={styles.menu} aria-label="Toggle Navigation"></button>
      <h2 className={styles.title}>{L('header.logo')}</h2>
      <Display if={!!user}>
        <div className={styles.profile}>
          <span>{user?.email}</span>
          (<a href={LOGOUT_ROUTE} onClick={handleLogoutClick}>{L('header.logout')}</a>)
        </div>
      </Display>
      <Display if={!user}>
        <div className={styles.profile}>
          <Link to={LOGIN_ROUTE}>{L('header.login')}</Link> / <Link to={SIGN_UP_ROUTE}>{L('header.signup')}</Link>
        </div>
      </Display>
    </div>
  );
}