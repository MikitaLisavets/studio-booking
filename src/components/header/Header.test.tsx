jest.mock('../../utils/locale');

import configureStore from 'redux-mock-store';
import Header from './Header';
import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';

function createComponent(newState = {}): ReactWrapper {
  const state = {
    user: null
  };

  const mockStore = configureStore();
  const store = mockStore({ ...state, ...newState });

  return mount(
    <StaticRouter>
      <Provider store={store}>
        <Header/>
      </Provider>
    </StaticRouter>
  );
}

describe('Header', () => {
  let component: unknown;

  describe('when user is empty', () => {
    beforeAll(() => {
      component = createComponent();
    });

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when user exists', () => {
    beforeAll(() => {
      component = createComponent({ user: { email: 'email@email.com' } });
    });

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
