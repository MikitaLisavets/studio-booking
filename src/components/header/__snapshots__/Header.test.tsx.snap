// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header when user exists renders correctly 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Header>
        <div
          className="header"
        >
          <button
            className="menu"
          />
          <h2
            className="title"
          >
            header.logo
          </h2>
          <Display
            if={true}
          >
            <div
              className="profile"
            >
              <span>
                email@email.com
              </span>
              (
              <a
                href="/logout"
                onClick={[Function]}
              >
                header.logout
              </a>
              )
            </div>
          </Display>
          <Display
            if={false}
          />
        </div>
      </Header>
    </Provider>
  </Router>
</StaticRouter>
`;

exports[`Header when user is empty renders correctly 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Header>
        <div
          className="header"
        >
          <button
            className="menu"
          />
          <h2
            className="title"
          >
            header.logo
          </h2>
          <Display
            if={false}
          />
          <Display
            if={true}
          >
            <div
              className="profile"
            >
              <Link
                to="/login"
              >
                <LinkAnchor
                  href="/login"
                  navigate={[Function]}
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                  >
                    header.login
                  </a>
                </LinkAnchor>
              </Link>
               / 
              <Link
                to="/signup"
              >
                <LinkAnchor
                  href="/signup"
                  navigate={[Function]}
                >
                  <a
                    href="/signup"
                    onClick={[Function]}
                  >
                    header.signup
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </Display>
        </div>
      </Header>
    </Provider>
  </Router>
</StaticRouter>
`;
